docs/data/date-pickers/date-time-picker/CustomDateTimePicker.tsx

```diff
@@ -1,58 +1,59 @@
 import * as React from 'react';
+import dayjs, { Dayjs } from 'dayjs';
 import AlarmIcon from '@mui/icons-material/Alarm';
 import SnoozeIcon from '@mui/icons-material/Snooze';
 import TextField from '@mui/material/TextField';
 import ClockIcon from '@mui/icons-material/AccessTime';
-import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
 import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
 import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
 import { MobileDateTimePicker } from '@mui/x-date-pickers/MobileDateTimePicker';
 import Stack from '@mui/material/Stack';
 
 export default function CustomDateTimePicker() {
-  const [clearedDate, setClearedDate] = React.useState<Date | null>(null);
-  const [value, setValue] = React.useState<Date | null>(
-    new Date('2019-01-01T18:54'),
-  );
+  const [dateWithNoInitialValue, setDateWithNoInitialValue] =
+    React.useState<Dayjs | null>(null);
+  const [dateWithInitialValue, setDateWithInitialValue] =
+    React.useState<Dayjs | null>(dayjs('2019-01-01T18:54'));
 
   return (
-    <LocalizationProvider dateAdapter={AdapterDateFns}>
+    <LocalizationProvider dateAdapter={AdapterDayjs}>
       <Stack spacing={3}>
         <DateTimePicker
           disableFuture
           hideTabs
           openTo="hours"
-          value={value}
+          value={dateWithInitialValue}
           onChange={(newValue) => {
-            setValue(newValue);
+            setDateWithInitialValue(newValue);
           }}
-          minDate={new Date('2018-01-01')}
+          minDate={dayjs('2018-01-01')}
           components={{
             LeftArrowIcon: AlarmIcon,
             RightArrowIcon: SnoozeIcon,
             OpenPickerIcon: ClockIcon,
           }}
-          minTime={new Date(0, 0, 0, 9)}
-          maxTime={new Date(0, 0, 0, 20)}
+          minTime={dayjs('2018-01-01T09:00')}
+          maxTime={dayjs('2018-01-01T20:00')}
           renderInput={(params) => (
             <TextField {...params} helperText="Hardcoded helper text" />
           )}
         />
         <MobileDateTimePicker
-          value={value}
+          value={dateWithInitialValue}
           onChange={(newValue) => {
-            setValue(newValue);
+            setDateWithInitialValue(newValue);
           }}
           label="With error handler"
           onError={console.log}
-          minDate={new Date('2018-01-01T00:00')}
-          inputFormat="yyyy/MM/dd hh:mm a"
+          minDate={dayjs('2018-01-01T00:00')}
+          inputFormat="YYYY/MM/DD hh:mm a"
           mask="___/__/__ __:__ _M"
```

Comment:
Just noticed one character is missing
```suggestion
          inputFormat="YYYY/MM/DD hh:mm a"
          mask="____/__/__ __:__ _M"
```